### ISSUE TITLE
bench(qwen38): publish exact M3 Pro P32/T64 parity close-out

### ISSUE THREAD

**Issue by anthony-chaudhary** [anthony-chaudhary/fak#9513]
## Parent context

M10 close-out child of #9430. Replaces the invalidly closed receipt boundary in #8697/#8972 for this plan phase; it does not reopen either issue and does not change #2723's broader Qwen3.6/Qwen2.5/MLX benchmark scope.

## Current state

The exact Qwen3.8 artifact and the pinned two-runtime oracle contract already exist. `internal/qwen38quantrun` binds fak-native versus llama.cpp b9828, strict execution roles, generated tokens/logits, P=32/T=64, temperature zero, three repetitions, timing, RSS, OS footprint, and hash-bound report/archive readback. What is still absent is one final committed bundle from post-M9 trunk and a witness that enforces the parent parity threshold. The existing oracle quality verdict does not enforce native decode >=95% of the comparator.

#8697 is closed only because its issue-referencing GEMV hill-climb commit was seen by the closure auditor. That valid slice measured 2.810041 tok/s, below the 6.617758 tok/s threshold, so it is not M10 completion. #8972 likewise closed without an accepted candidate/comparator pair.

## Core through-line

For an operator running the exact Qwen3.8-27B Q4_K_M artifact on the sanctioned M3 Pro, publish the final matched native/comparator result after M6-M9 without changing execution engines or mixing artifacts.

## Problem frame

- **Centrality:** Core — this is the terminal parity verdict for the #9430 Mac performance program.
- **P1 managed context / quality:** Freeze the artifact, prompt hash, P32/T64 shape, temperature zero, generated tokens, logits, three repetitions, and clean session state for both same-artifact arms.
- **P2 net-true efficiency:** Retain TTFT, prefill and decode seconds/tok/s, RSS, OS footprint, load/setup, commands, revisions, binary identity, and every admitted repetition; no microbenchmark substitutes.
- **P3 bounded adaptation:** The product arm remains `engine=inkernel backend=metal forward_path=metal/qwen35-hybrid-session-v1 q4k=true fallback_active=false comparator_only=false`. llama.cpp is comparator-only. Any fallback or identity drift invalidates the bundle.
- **P4 integrated operations:** One public-safe immutable bundle, replay command, hash readback, source/module revisions, hardware-quiescence evidence, and explicit PASS/HOLD result make the close-out auditable and rerunnable.

## Value frame

- **For:** M3 Pro operators and maintainers deciding whether fak-native Qwen3.8 has reconverged with the pinned Mac comparator.
- **Problem:** the repository has enabling gains and partial receipts, but no post-M9 same-artifact parity bundle enforcing the >=95% completion threshold.
- **Today:** the closest committed comparison is approximate, and prior receipt issues were closed without a quality-complete accepted pair.
- **Better because:** one raw-evidence-derived verdict prevents issue closure, external runtime execution, or mixed artifact/model rows from masquerading as fak-native parity.
- **Witness:** one test replays and validates the complete immutable bundle, then recomputes deterministic quality and both throughput ratios from raw repetitions.

## Exact operating envelope

- Host: Apple M3 Pro, 18 GPU cores, 36 GiB unified memory.
- Model: `unsloth/Qwen3.8-27B-GGUF@f1bfb127c64f7072bdd2cad55f258b9c8b2910fe`.
- Artifact: Q4_K_M, 17,106,775,008 bytes, SHA-256 `7e78da5d7e3ae28d178121f58646953305f3e5bd3cb46f4a75584e8b6c6fe169`.
- Shape: P=32, T=64, temperature=0, exactly three accepted repetitions per same-artifact arm.
- Native identity: `inkernel / metal / metal/qwen35-hybrid-session-v1 / q4k=true / fallback=false / comparator_only=false`.
- Comparator identity: llama.cpp build b9828, revision `ebd048fc5e4b43ec4e0b4abe0b9bf66e1724dad0`, MIT, `metal / llama.cpp/metal / q4k=true / fallback=false / comparator_only=true`.
- Comparator binary: record and verify its SHA-256; the previously observed b9828 binary hash is `12df97ffa9d48545e96cd3237a71f78efd1cc0222f971cbd65f7ab57e793b128`.
- Installed Homebrew b8200 is not an accepted substitute.

## Working spine

Post-M9 clean trunk and quiescent Mac -> exact fak-native and pinned llama.cpp commands -> three serialized same-artifact repetitions each -> qwen38 oracle report/archive fold -> one docs-only bundle -> independent readback recomputes quality and >=95% arithmetic/geometric throughput ratios -> PASS or honest HOLD.

## In scope

Capture and commit one docs-only, public-safe M10 bundle containing exact commands, revisions, binary/artifact/prompt hashes, runtime identities, three raw repetitions per same-artifact arm, tokens/logits, timing/memory/profile evidence, oracle report/archive, checksums, and one deterministic replay test. Publish PASS only when all gates below pass; otherwise publish the same complete bundle as HOLD.

MLX may appear only as an optional separately typed observation in this same bundle. It must name its runtime revision and its own artifact hash. Unless that hash equals the exact GGUF hash above, mark it `comparability=equivalent-model-only` and exclude it from quality equality and the 95% parity denominator. Absence of MLX does not weaken or alter the same-artifact fak-native/llama.cpp verdict.

## Out of scope

No kernel, model, Metal, scheduler, serving, benchmark-schema, or command implementation changes; no runtime fallback; no tuning of either engine; no new MLX adapter; no generic three-way benchmark; no Qwen3.6/Qwen2.5 row; no plan/index/front-door update in this issue; and no extra optimization queue after the measured verdict.

If current committed tooling cannot capture or validate a required field, stop and file a separate prerequisite tooling issue rather than expanding this docs-only leaf.

## Dependencies and hardware admission

- Hard dependencies: M6 paged hybrid state, M7 exact-prefix reuse, M8 bounded chunked-prefill scheduling, and M9 resident hybrid co-batching under #9430 must each have their accepted keep/reject receipts.
- Source: build from clean pushed trunk containing those reconciled outcomes; bind `vcs.revision`, `vcs.modified=false`, binary hash, and changed modules as `module@rev`.
- Hardware: dispatch only after the active exact-model serve/campaign and every machine-wide GPU lease have ended. Acquire one continuous exclusive Mac GPU lease for both serialized arms. Record quiescence/ownership evidence; concurrent Metal inference invalidates all timings.
- File lane: docs only. Do not acquire or edit `internal/**`, `cmd/**`, or `experiments/**`.

## Exact path boundary

Only:

`docs/_witnesses/issue-9513-qwen38-m10-parity/**`

Expected bundle members:

- `README.md`
- `doc.go`
- `oracle-config.json`
- `fak-adapter.json`
- `llama-adapter.json`
- fak and llama campaign report/raw-archive pairs
- fak and llama v2 measurement JSON
- `oracle-report.json`
- `oracle-archive.json`
- `profile.json`
- `profile.receipt.json`
- optional `mlx-observation.json`
- `SHA256SUMS`
- `witness_test.go`

## Acceptance gate

The bundle is accepted as one unit only when one raw-evidence readback jointly proves the exact artifact bytes/hash, host, prompt hash, P32/T64 temperature-zero three-repetition envelope, clean source, commands, comparator revision/license, and both binary hashes; 64 nonempty generated tokens plus finite logits, positive TTFT/prefill/decode timing and rates, RSS, and OS footprint in every repetition; deterministic tokens across repetitions and equality between same-artifact arms with logits inside the frozen tolerance; exact fak-native identity with zero fallback and comparator-only llama.cpp identity; arithmetic-mean and geometric-mean native/comparator decode ratios each >=0.95; hash-bound secret-scrubbed profiles, raw archives, reports, configs, and checksums; and optional MLX evidence separately typed and excluded unless it proves the exact artifact hash.

Any failed, missing, interrupted, mixed-envelope, or concurrent sample produces HOLD and no M10 KEEP credit.

## Done condition

One immutable docs-only bundle at the exact path validates from raw evidence on trunk and records the final PASS or honest HOLD for the exact M3 Pro/Qwen3.8 P32/T64 fak-native-versus-pinned-llama.cpp parity gate, with optional MLX evidence classified separately as specified.

## Witness

`go test ./docs/_witnesses/issue-9513-qwen38-m10-parity -run '^TestMatchedParityReceipt$' -count=1 -v`

The test must recompute artifact/config/report/archive hashes, strict runtime identities, token/logit parity, repetition cardinality, arithmetic mean, geometric mean, and both >=0.95 ratios from raw committed evidence.

## Replay gate

```sh
go run ./cmd/qwen38campaign --oracle \
  --config docs/_witnesses/issue-9513-qwen38-m10-parity/oracle-config.json \
  --corpus docs/benchmarks/qwen38-quant/corpus.json \
  --report /tmp/m10-report.json \
  --archive /tmp/m10-archive.json

go test ./docs/_witnesses/issue-9513-qwen38-m10-parity \
  -run '^TestMatchedParityReceipt$' -count=1 -v
```

The regenerated report/archive must byte-match or hash-match the committed derived artifacts.

## Work unit

Leaf. One deliverable, one witness, one docs-only commit.

## Work scale

S1, estimated 3 steps: quiescent capture, deterministic fold, exact-path commit.

## Lane

`docs`

## Gold-plating boundary

Stop after the final exact-envelope bundle reports PASS or HOLD. Do not tune, generalize schemas, add an MLX runtime path, update broader benchmark catalogs, or turn the result into another optimization phase.

## Closure binding

Close only from one signed commit citing `Fixes #9513`, carrying `(fak docs)`, touching only the exact witness directory, and passing the replay witness. Closing this child does not independently close #9430; the parent completion audit decides M10 and the 10/10 hero metric.

## Why this is next

The mechanism and parity prerequisites are already landed, but the portfolio still lacks one immutable exact M3 Pro P32/T64 close-out that binds the current fak-native Qwen3.8 Metal engine, artifact, quality, memory, and timing envelope. Closing this evidence leaf turns the delivered path into an auditable KEEP/REJECT decision without widening product scope.

## Likely files

- `docs/_witnesses/issue-9513-qwen38-m3pro-p32-t64/**`
- the existing close-out index or benchmark authority row that consumes that immutable receipt, only if required by the current witness contract

**Comment by anthony-chaudhary** [comment]
Six-hour Qwen run audit at origin/main `e5af55ed9`: keep OPEN. The required `docs/_witnesses/issue-9513-qwen38-m10-parity/**` bundle, raw-readback test, report/archive, SHA256SUMS, profile pair, replay binding, and `Fixes #9513` commit do not exist. Existing #8697 evidence is a genuine but sub-target GEMV result and #8972 records a REFUSE_UNSAFE campaign with zero accepted matched samples; neither satisfies the exact 3+3 P32/T64 parity contract. Parent #9430 remains 0/10 KEEP and the authoritative registry still places M6-M9 and M10 behind dependencies. Next check: `git show origin/main:docs/benchmarks/NATIVE-PERFORMANCE-CURRENT.md | rg "mac\\.m(6|7|8|9|10)-"`; only after M6-M9 clear should the M10 hardware capture and exact `TestMatchedParityReceipt` run. `CLAIM_TEST_UNRUN`; no parity/performance claim.

**Comment by anthony-chaudhary** [comment]
Claiming epic #10193 ranked row 7 jointly with the companion issue. Scope is exact CUDA/Metal checkpoint reconvergence only. I will audit duplicate coverage, exact Qwen3.8-27B Q4_K_M matched baseline/current evidence, the immutable M10 witness bundle, and M6-M9 dependency clearance. No speculative work, row 6 #8324, Mac prefill reordering, service-throughput, or speculative-campaign changes; no performance claim without engine=fak-native, zero fallback, strict quality, one-variable matched A/B, and inclusive request-boundary accounting.

**Comment by anthony-chaudhary** [comment]
**HOLD_DEPENDENCIES_AND_MATCHED_EVIDENCE_MISSING** — epic #10193 ranked row 7 remains open and is demoted to `priority/P2` + `gen/next` with `gated/hardware`.

Duplicate reconciliation found no separate tracker superseding #9513. At `origin/main` `30a60f4dbadca500be708111b1ca6d9a2d3bc10a`, `docs/benchmarks/NATIVE-PERFORMANCE-CURRENT.md` still marks M6, M7, M8, M9, and M10 `waiting-dependency`; #9430 still has M6–M10 incomplete. Dependency state is not cleared: #8395, #1912, #9074, and #9075 remain open; closure of #9076 and #9066 does not supply accepted keep/reject receipts for every M6–M9 arm.

The immutable `docs/_witnesses/issue-9513-qwen38-m10-parity/` bundle does not exist on trunk. The required test therefore fails closed before execution:

```text
go test ./docs/_witnesses/issue-9513-qwen38-m10-parity -run '^TestMatchedParityReceipt$' -count=1 -v
stat .../docs/_witnesses/issue-9513-qwen38-m10-parity: directory not found
FAIL ... [setup failed]
```

This is missing-evidence/dependency status, not parity evidence. **No M10, parity, or performance credit is claimed.** Creating a placeholder bundle now would weaken the fail-closed contract, so no repository file is added.

After M6–M9 each have accepted isolated keep/reject receipts, acquire the exclusive exact Apple M3 Pro (18 GPU cores, 36 GiB) lease and run the exact Qwen3.8-27B Q4_K_M P=32/T=64 matched campaign: native `inkernel / metal / metal/qwen35-hybrid-session-v1`, `q4k=true`, `engine=fak-native`, `fallback=false`, strict quality, exactly three accepted repetitions per arm, versus the pinned comparator only, with inclusive request-boundary accounting. Then replay:

```bash
go run ./cmd/qwen38campaign --oracle \
  --config docs/_witnesses/issue-9513-qwen38-m10-parity/oracle-config.json \
  --corpus docs/benchmarks/qwen38-quant/corpus.json \
  --report /tmp/m10-report.json \
  --archive /tmp/m10-archive.json

go test ./docs/_witnesses/issue-9513-qwen38-m10-parity \
  -run '^TestMatchedParityReceipt$' -count=1 -v
```

**Comment by anthony-chaudhary** [comment]
**HOLD_PREREQUISITES_NOT_CLEARED — rechecked 2026-08-30 from current `main`.**

I attempted the sanctioned M10 entry path rather than stopping at the first failure. The exact close-out still cannot be run or honestly committed:

- `docs/benchmarks/NATIVE-PERFORMANCE-CURRENT.md` still marks `mac.m6-paged-hybrid-state`, `mac.m7-prefix-reuse`, `mac.m8-chunked-prefill`, `mac.m9-resident-cobatching`, and `mac.m10-parity-reconvergence` as `waiting-dependency`. No `docs/_witnesses/issue-8395*`, `issue-1912*`, or `issue-9074*` accepted dependency bundle exists.
- The required post-M9 clean trunk condition is false: the shared checkout has 45 peer-owned dirty paths, and `main` moved during the audit. I did not sweep, reset, or capture a mixed source state.
- The exact Qwen3.8 artifact is present and hashes to `7e78da5d7e3ae28d178121f58646953305f3e5bd3cb46f4a75584e8b6c6fe169`. The pinned llama.cpp b9828 binary was also located at `/Users/anthony/llama-b9828/llama-b9828/llama-cli` (`version: 9828`, commit `ebd048fc5`), so local hardware/artifact availability is not the terminal blocker.
- The named replay adaptation was attempted exactly. `go run ./cmd/qwen38campaign --oracle --config docs/_witnesses/issue-9513-qwen38-m10-parity/oracle-config.json --corpus docs/benchmarks/qwen38-quant/corpus.json --report /tmp/m10-report.json --archive /tmp/m10-archive.json` fails because the prerequisite-bound config/bundle does not exist.
- The acceptance witness still fails closed because `docs/_witnesses/issue-9513-qwen38-m10-parity` does not exist. The origin lane commands also fail because `internal/qwen38bench` is not a package in this checkout; the existing implementation package is `internal/qwen38quantrun`.
- `dos lease-lane acquire --lane qwen38bench ...` reports that `qwen38bench` is not configured and auto-picks unrelated `cdb`; I left no such lease held. The bounded harness self-check passed with a typed `fak.microharness.usage.v1` completed receipt.

No placeholder bundle was added and no parity/performance credit is claimed. Creating one before M6–M9 receipts would violate this issue's fail-closed ordering and could only fabricate a green-looking result.

**Smallest next checkable step:** close #8395/#1912/#9074 with accepted isolated M6–M9 keep/reject receipts and update the registry rows; then obtain a clean/quiescent `main`, acquire a real `hardware+docs`/qwen38 lane, build the fak-native binary from that exact HEAD, run the pinned b9828 comparator and fak-native arms serially for exactly 3+3 P32/T64 repetitions against artifact hash `7e78…e169`, fold the oracle report/archive, and run `TestMatchedParityReceipt`.

**Comment by anthony-chaudhary** [comment]
Completed with exact M3 Pro P32/T64 parity close-out bundle and replay test.
